src/GameData.ts

```diff
@@ -0,0 +1,265 @@
+        });
+    }
+
+    public async reloadData() {
+        this.champData = await this.loadChampionData();
+        this.perkData = await this.loadPerkData();
+        this.itemData = await this.loadItemData();
+        console.log("Game data loaded!");
+    }
+
+    public async loadChampionData(): Promise<ChampionData[]> {
+        const returnData: ChampionData[] = [];
+
+        const championDataUrl = "https://raw.communitydragon.org/latest/plugins/rcp-be-lol-game-data/global/default/v1/champion-summary.json";
+        const champions = await (await fetch(championDataUrl)).json();
+        champions.filter((c: any) => c.id > 0)
+            .forEach((c: any) => returnData.push({
+                id: c.id,
+                name: c.name,
+                key: c.alias,
+                skins: [],
+            }));
+
+        const skinDataUrl = "https://raw.communitydragon.org/latest/plugins/rcp-be-lol-game-data/global/default/v1/skins.json";
+        const skins = await (await fetch(skinDataUrl)).json() as SkinData[];
+
+        (champions as ChampionData[]).map(c => c.id)
+            .filter(id => id > 0)
+            .map(i => i.toString())
+            .forEach(id => {
+                const skinKeys = Object.keys(skins)
+                    .filter(k => k.startsWith(id))
+                    .filter(k => (k.length - id.length) === 3)
+                    .map(k => +k);
+
+                skinKeys.forEach(key => {
+                    const data = skins[key];
+                    returnData.filter(r => r.id === +id)[0].skins.push({ ...data });
+                });
+            });
+
+        return returnData;
+    }
+
+    public async loadPerkData(): Promise<PerkData[]> {
+        const perkDataUrl = "https://raw.communitydragon.org/latest/plugins/rcp-be-lol-game-data/global/default/v1/perks.json";
+        return await (await fetch(perkDataUrl)).json() as PerkData[];
+    }
+
+    public async loadItemData(): Promise<ItemData[]> {
+        const returnData: ItemData[] = [];
+
+        const itemDataUrl = "https://raw.communitydragon.org/latest/plugins/rcp-be-lol-game-data/global/default/v1/items.json";
+        const items = await (await fetch(itemDataUrl)).json();
+        items.forEach((c: ItemData) => returnData.push({ ...c }));
+        returnData.forEach((c: ItemData) => { c.to = []; c.from = []; });
+
+        items.forEach((i: any) => {
+            const item = returnData.filter(r => r.id === +i.id)[0];
+
+            i.from.forEach((f: number) => {
+                const otherItem = returnData.filter(r => r.id === f)[0];
+                if (otherItem !== undefined) {
+                    item.from.push(otherItem.name);
+                }
+            });
+
+            i.to.forEach((f: number) => {
+                const otherItem = returnData.filter(r => r.id === f)[0];
+                if (otherItem !== undefined) {
+                    item.to.push(otherItem.name);
+                }
+            });
+        });
+
```

Comment:
```js
for (const item of items) {
    const complex = returnData.find(x => x.id === item.id)!;
   
    complex.from = item.from.map(x => returnData.find(y => y.id === x)!).filter(x => x).map(x => x.name);
    complex.to = item.to.map(x => returnData.find(y => y.id === x)!).filter(x => x).map(x => x.name);
}
```

Reply:
Hey I know one liners look cool but have you tried `making your code readable`?